### PR TITLE
Add modules for open-source hosted applications

### PR DESCRIPTION
**Pull request by jonmv**
@gjoranv please review. 

The different test profiles in the pom are only needed for running the tests locally, but I guess it's nice to have that option ... not that anything at all is supported yet, but ... :)

### CODE REVIEW

File: tenant-base/pom.xml [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+    <version>6-SNAPSHOT</version>
+    <name>Hosted Vespa tenant base</name>
+    <description>Parent POM for all hosted Vespa applications.</description>
+    <url>https://github.com/vespa-engine</url>
+    <packaging>pom</packaging>
+
+    <properties>
+        <vespaversion>${project.version}</vespaversion>
+        <target_jdk_version>1.8</target_jdk_version>
+        <compiler_plugin_version>3.6.1</compiler_plugin_version>
```
Comment (gjoranv) [inline tenant-base/pom.xml]: Update to 3.8.0

File: tenant-base/pom.xml [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+
+            <dependency>
+                <groupId>com.yahoo.vespa</groupId>
+                <artifactId>container-dependency-versions</artifactId>
+                <version>${vespaversion}</version>
+                <type>pom</type>
+                <scope>import</scope>
+            </dependency>
+
+            <dependency>
```
Comment (gjoranv) [inline tenant-base/pom.xml]: Please remove. Dep hack for oath apps.

File: tenant-base/pom.xml [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+            </plugin>
+
+            <plugin>
+                <groupId>com.yahoo.vespa</groupId>
+                <artifactId>bundle-plugin</artifactId>
+                <version>${vespaversion}</version>
+                <extensions>true</extensions>
+            </plugin>
+
+            <plugin>
```
Comment (gjoranv) [inline tenant-base/pom.xml]: Remove

**Review by gjoranv — APPROVED** [review]

**Comment by jonmv** [comment]
FYI the `bundle-plugin` dependency was required to know about the `container-plugin` packaging :)

**Comment by jonmv** [comment]
Oops, forgot to check the PR build.

**Comment by gjoranv** [comment]
> FYI the bundle-plugin dependency was required to know about the container-plugin packaging :)

true, that's what the `<extension>true</>` is about

**Pull request merged**